plugins/tech-insights-backend/src/service/fact/FactRetrieverRegistry.ts

```diff
@@ -50,27 +55,31 @@ export class DefaultFactRetrieverRegistry implements FactRetrieverRegistry {
       );
     }
     this.retrievers.set(registration.factRetriever.id, registration);
+    return Promise.resolve();
```

Comment:
Make the methods `async` instead, that way errors won't be thrown synchronously as well